lib/ecto/repo/supervisor.ex

```diff
@@ -25,7 +26,22 @@ defmodule Ecto.Repo.Supervisor do
     case repo_init(type, repo, config) do
       {:ok, config} ->
         {url, config} = Keyword.pop(config, :url)
-        {:ok, Keyword.merge(config, parse_url(url || ""))}
+        url_config = parse_url(url || "")
+
+        # if config already has `ssl` set, e.g.: `ssl: [cacertfile: "/path/to/file"]`
+        # the URL's `ssl=true` parameter is dropped
+        url_config =
+          if is_list(config[:ssl]) and url_config[:ssl] == true do
+            Logger.warning(
+              "Ignoring `ssl=true` parameter in URL because `ssl` is already set in the configuration: #{inspect(config[:ssl])}"
```

Comment:
```suggestion
        url_config =
          if is_list(config[:ssl]) and url_config[:ssl] == true do
            Logger.warning(
              "ignoring `ssl=true` parameter in URL because `ssl` is already set in the configuration: #{inspect(config[:ssl])}"
```